subsys/net/ip/net_context.c

```diff
@@ -338,11 +343,15 @@ int net_context_put(struct net_context *context)
 	context->send_cb = NULL;
 
 	if (net_tcp_put(context) >= 0) {
-		return 0;
+		goto out;
```

Comment:
Could we here follow net_context_bind() below, and unlock and return 0? Then the functions would be cosistent, at least for these two functions.

Reply:
I would rather keep the out: label here so that we have the unlock called at the end of the function. Note that the out: label is called also from offloading case few lines above.